engine/apps/api/views/user.py

```diff
@@ -234,9 +234,27 @@ def schedules_with_oncall_users(self):
         """
         return get_oncall_users_for_multiple_schedules(self.request.user.organization.oncall_schedules.all())
 
+    def _get_is_currently_oncall_query_param(self) -> str:
+        return self.request.query_params.get("is_currently_oncall", "").lower()
+
+    def _is_currently_oncall_request(self) -> bool:
+        return self._get_is_currently_oncall_query_param() in ["true", "false"]
+
+    def _is_short_oncall_request(self) -> bool:
+        return self.request.query_params.get("short", "true").lower() == "false"
+
+    def _is_currently_oncall_or_short_request(self) -> bool:
+        return self._is_currently_oncall_request() or self._is_short_oncall_request()
+
     def get_serializer_context(self):
         context = super().get_serializer_context()
-        context.update({"schedules_with_oncall_users": self.schedules_with_oncall_users})
+        context.update(
+            {
+                "schedules_with_oncall_users": self.schedules_with_oncall_users
+                if self._is_currently_oncall_or_short_request()
+                else {}
```

Comment:
main change, _only_ set `context["schedules_with_oncall_users"]` for certain requests. Previously we were _always_ setting it even if the serializer didn't need it (`UserLongSerializer` is the only serializer that needs this [here](https://github.com/grafana/oncall/pull/3220/files#diff-30309629484ad28e6fe09816e1bd226226d652ea977b6f3b6775976c729bf4b5R271-R272)).